x-pack/plugins/ml/public/application/components/ml_page/notifications_indicator.tsx

```diff
@@ -5,22 +5,118 @@
+import { useStorage } from '../../contexts/storage';
+import { ML_NOTIFICATIONS_LAST_CHECKED_AT } from '../../../../common/types/storage';
+
+const NOTIFICATIONS_CHECK_INTERVAL = 60000;
 
 export const NotificationsIndicator: FC = () => {
+  const {
+    services: {
+      mlServices: { mlApiServices },
+    },
+  } = useMlKibana();
+  const { euiTheme } = useEuiTheme();
+  const [lastCheckedAt] = useStorage(ML_NOTIFICATIONS_LAST_CHECKED_AT);
+
+  const lastCheckedAt$ = useAsObservable(lastCheckedAt);
+
+  const [notificationsCounts, setNotificationsCounts] = useState<NotificationsCountResponse>();
+
+  useEffect(function startPollingNotifications() {
+    const subscription = combineLatest([
+      lastCheckedAt$.pipe(filter((v): v is number => !!v)),
+      timer(0, NOTIFICATIONS_CHECK_INTERVAL),
+    ])
+      .pipe(
+        switchMap(([lastChecked]) =>
+          mlApiServices.notifications.countMessages$({ lastCheckedAt: lastChecked })
+        ),
+        catchError((error) => {
+          // Fail silently for now
+          return of({} as NotificationsCountResponse);
+        })
+      )
+      .subscribe((response) => {
+        setNotificationsCounts(response);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const errorsAndWarningCount =
+    (notificationsCounts?.error ?? 0) + (notificationsCounts?.warning ?? 0);
+  const hasUnread = notificationsCounts && Object.values(notificationsCounts).some((v) => v > 0);
+
   return (
-    <EuiFlexGroup alignItems={'center'} gutterSize={'xs'}>
+    <EuiFlexGroup alignItems={'center'} gutterSize={'s'}>
       <EuiFlexItem grow={false}>
         <FormattedMessage
           id="xpack.ml.navMenu.notificationsTabLinkText"
           defaultMessage="Notifications"
         />
       </EuiFlexItem>
-      {false ? (
+      {errorsAndWarningCount ? (
+        <EuiFlexItem grow={false}>
+          <EuiToolTip
+            position="right"
+            content={
+              <FormattedMessage
+                id="xpack.ml.notificationsIndicator.errorsAndWarningLabel"
+                defaultMessage="There {count, plural, one {is # notification} other {are # notification}} with the error or warning level"
+                values={{ count: errorsAndWarningCount }}
+              />
+            }
+          >
+            <EuiNotificationBadge>{errorsAndWarningCount}</EuiNotificationBadge>
+          </EuiToolTip>
+        </EuiFlexItem>
+      ) : null}
+      {!errorsAndWarningCount && hasUnread ? (
         <EuiFlexItem grow={false}>
-          <EuiNotificationBadge>{0}</EuiNotificationBadge>
+          <EuiToolTip
+            position="right"
+            content={
+              <FormattedMessage
+                id="xpack.ml.notificationsIndicator.unreadLabel"
+                defaultMessage="You have unread notifications"
+              />
+            }
+          >
+            <svg
```

Comment:
Maybe you can use `EuiHealth` here instead of custom SVG? https://elastic.github.io/eui/#/display/health

Reply:
Replaced in https://github.com/elastic/kibana/pull/140944/commits/686bfd00d2135dc34e9eaac737420d7ae0def722